rdkPlugins/OOMCrash/source/OOMCrashPlugin.cpp

```diff
@@ -0,0 +1,200 @@
+}
+
+/**
+ *  * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ *   */
+bool OOMCrash::postInstallation()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+	return false;
+    }
+
+    const std::string source = mContainerConfig->rdk_plugins->oomcrash->data->path;
+    const std::string dest = mContainerConfig->rdk_plugins->oomcrash->data->path;
+    
+    if (!mUtils->addMount(source, dest, "bind", {"bind", "nodev","nosuid", "noexec" }))
+    {
+        AI_LOG_WARN("failed to add mount %s", source.c_str());
+	return false;
+    }
+    
+    AI_LOG_INFO("This hook is running for container with hostname %s", mUtils->getContainerId().c_str());
+    return true;
+}
+
+/**
+ * @brief Dobby Hook - Run in host namespace when container terminates
+ */
+bool OOMCrash::postHalt()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+        return false;
+    }
+    
+    if(mUtils->exitStatus == 0)
+        checkForOOM();
+    
+    AI_LOG_INFO("This hook is running for container with hostname %s", mUtils->getContainerId().c_str());
+    return true;
+}
+
+// End hook methods
+
+/**
+ * @brief Should return the names of the plugins this plugin depends on.
+ *
+ * This can be used to determine the order in which the plugins should be
+ * processed when running hooks.
+ *
+ * @return Names of the plugins this plugin depends on.
+ */
+
+std::vector<std::string> OOMCrash::getDependencies() const
+{
+    std::vector<std::string> dependencies;
+    const rt_defs_plugins_oom_crash* pluginConfig = mContainerConfig->rdk_plugins->oomcrash;
+
+    for (size_t i = 0; i < pluginConfig->depends_on_len; i++)
+    {
+        dependencies.push_back(pluginConfig->depends_on[i]);
+    }
+
+    return dependencies;
+}
+
+/**
+ * @brief Read cgroup file.
+ *
+ * @return true on success.
+ */
+
+bool OOMCrash::readCgroup(unsigned long *val)
+{
+    std::string path = "/sys/fs/cgroup/memory/" + mUtils->getContainerId() + "/memory.failcnt";
+
+    FILE *fp = fopen(path.c_str(), "r");
+    if (!fp)
+    {
+	if (errno != ENOENT)
+	    AI_LOG_ERROR("failed to open '%s' (%d - %s)", path.c_str(), errno, strerror(errno));
+
+	return false;
+    }
+
+    char* line = nullptr;
+    size_t len = 0;
+    ssize_t rd;
+
+    if ((rd = getline(&line, &len, fp)) < 0)
+    {
+        if (line)
+	        free(line);
+	    fclose(fp);
+	    AI_LOG_ERROR("failed to read cgroup file line (%d - %s)", errno, strerror(errno));
+	    return false;
+    }
+
+    *val = strtoul(line, nullptr, 0);
+
+    fclose(fp);
+    free(line);
+
+    return true;
+}
+
+/**
+ * @brief Check for Out of Memory.
+ *
+ * @return void.
+ */
+
+void OOMCrash::checkForOOM()
+{
+    unsigned long failCnt;
+    
+    if (readCgroup(&failCnt) && (failCnt > 0))
+    {
+	    AI_LOG_ERROR("memory allocation failure detected in %s container, likely OOM (failcnt = %lu)", mUtils->getContainerId().c_str(), failCnt);
```

Comment:
This shouldn't be error, as this plugin is for catching oom finding one is "normal". Move it to either warning. Errors are reserved for things that will disturb the functionality of current code. We would get error that container exited either way in different place, but at least we will know that it wasn't due to oom plugin.